### ISSUE TITLE
Assistance Needed: Code Execution Issue on CAVE Dataset

### ISSUE THREAD

**Issue by LTTdouble** [HuQ1an/GELIN_TGRS#1]
I trust this email finds you well. I am one of the readers who recently delved into your work on GELIN and attempted to run your provided code on the 4X dataset from CAVE. Firstly, I want to express my sincere gratitude for providing such exceptional work and generously sharing your research findings.

However, while attempting to execute your code (using the parameters you provided), I encountered a small issue that I hope to seek your assistance with. As per your paper and code, you described the grouping of the dataset, such as in the case of the CAVE dataset, where there are 31 spectral bands. Each group comprises 8 bands with a 2-band overlap, and in cases where the last group contains fewer than 8 bands, it should encompass the last 8 bands. However, when running the code, an error occurred, as depicted in the attached image, where the last group does not consist of 8 bands.
![1](https://github.com/HuQ1an/GELIN_TGRS/assets/130349905/98c5ec28-c8f8-4040-85e6-ca7ba15a711e)
![2](https://github.com/HuQ1an/GELIN_TGRS/assets/130349905/9a130f5e-6549-42fb-be94-9d39c49208b2)

I am eager to understand how you addressed this issue. Are there specific methods or techniques that allow the program to run correctly even when the last group has fewer than 8 bands? If you could share some insights or suggestions on resolving this issue, I would greatly appreciate it.

Looking forward to your response and once again, thank you for your invaluable time and outstanding work.

**Comment by HuQ1an** [comment]
I haven't encountered the issue you mentioned. When grouping hyperspectral images, if the number of bands in the last group is less than n_subs, then the group will include the last n_subs bands counted backward from the last band. Perhaps you can check if your hyperparameters such as n_subs, n_ovls, n_colors, etc., are set correctly. 
And moreover, you can also print the value of 'start_idx' and 'end_idx' to check if it is the same as ours:
start_idx: [0, 6, 12, 18, 23]
end_idx: [8, 14, 20, 26, 31]

**Comment by LTTdouble** [comment]
Thank you very much for your response and the reminders.  I have been using the parameters you provided in both the mains.py file and the demo.sh script.

![1](https://github.com/HuQ1an/GELIN_TGRS/assets/130349905/eff83ead-423e-442e-af67-a8fd8885c73a)
![2](https://github.com/HuQ1an/GELIN_TGRS/assets/130349905/c2685f2a-62ee-4630-b35b-6bb582e96a4a)

We have printed the values of start_idx and end_idx, which are [0, 6, 12, 18, 24] and [8, 14, 20, 26, 32] respectively. However, the shape of the input image is (8, 31, 32, 32).

We have noticed that the last group does not match the values you provided. We will further investigate the data loading process. Once again, I appreciate your patient guidance and valuable suggestions.
